android/engine/src/main/java/org/smartregister/fhircore/engine/configuration/ConfigurationRegistry.kt

```diff
@@ -81,6 +93,24 @@ constructor(
     return configsJsonMap.getValue(configType.name).decodeResourceFromString()
   }
 
+  /**
+   * Retrieve translation configuration for the provided [bundleName]. The Bundle value is retrieved
+   * from [configsJsonMap] can be directly converted to a ResourceBundle.
+   */
+  fun retrieveResourceBundleConfiguration(bundleName: String): ResourceBundle? {
+    return try {
+      PropertyResourceBundle(configsJsonMap[bundleName]!!.byteInputStream())
+    } catch (e: NullPointerException) {
+      if (bundleName.contains("_")) {
+        return retrieveResourceBundleConfiguration(
+          bundleName.substring(0, bundleName.lastIndexOf('_'))
+        )
+      } else {
+        null
+      }
+    }
+  }
```

Comment:
```suggestion
  fun retrieveResourceBundleConfiguration(bundleName: String): ResourceBundle? {
    val resourceBundle = configsJsonMap[bundleName]
    if (resourceBundle != null) {
      return PropertyResourceBundle(resourceBundle.byteInputStream())
    }
    if (bundleName.contains("_")) {
      return retrieveResourceBundleConfiguration(
        bundleName.substring(0, bundleName.lastIndexOf('_'))
      )
    }
    return null
  }
```

Reply:
The **try-catch** block is unnecessary. `configsJsonMap[bundleName]!!.byteInputStream()` will force the `NullPointerException`.